Documentation/ExtensionArchitecture/Tutorials/Tea/DirectoryStructure.rst

```diff
@@ -0,0 +1,189 @@
+
+In most cases the **view** is handled by classes provided by the framework
+and configured via templating. Therefore there is no folder for the view as a
+whole.
+
+Additional logic needed for the view can be provided by **ViewHelpers** and
+should be stored in the according folder.
+
+..  note::
+    ViewHelpers are a feature of the Fluid templating engine. See also the
+    chapter :ref:`fluid-custom-viewhelper`.
+
+Directory :file:`Configuration`
+-------------------------------
+
+See also the general chapter on the folder :ref:`extension-configuration-files`.
+
+The folder :file:`Configuration` contains several subfolders:
+
+..  code-block:: none
+    :caption: Directory structure of EXT:tea
+
+    $ tree typo3conf/ext/tea
+    ├── Configuration
+        ├── FlexForms
+        ├── TCA
+        |   └── Overrides
+        ├── TsConfig
+        |   ├── Page
+        |   └── User
+        ├── TypoScript
+        |   ├── constants.typoscript
+        |   └── setup.typoscript
+        └──  Services.yaml
+
+:file:`Configuration/FlexForms`
+    Contains the configuration of additional input fields to
+    configure the plugin.
+:file:`Configuration/TCA`
+    Contains the TYPO3 configuration array (TCA) as PHP arrays.
+:file:`Configuration/TCA/Overrides`
+    Can be used to extend the TCA of other extensions. They can be extended
+    by direct array manipulation or (preferred if possible) by calls to
+    API functions.
+:file:`Configuration/TsConfig`
+    Contains configurations for the TYPO3 backend on page or user level in
+    the syntax TypoScript. This extension does not feature TS
+:file:`Configuration/TypoScript`
+    Contains configurations for the TYPO3 frontend. In some contexts the
+    configuration contained here is also considered in the backend.
+:file:`Configuration/Services.yaml`
+    Is used to configure technical aspects of the extension including
+    automatic wiring, automatic configuration and options for dependency
+    injection.
+
+Directory :file:`Documentation`
+-------------------------------
+
+The folder :file:`Documentation` contains the files from which the
+documentation is rendered. See :ref:`extension-files-documentation`.
+
+Directory :file:`Resources`
+---------------------------
+
+See also the general chapter on the folder :ref:`extension-Resources`.
+
+The folder :file:`Resources` contains two sub folders that are
+further divided:
+
+..  code-block:: none
+    :caption: Directory structure of EXT:tea
+
+    $ tree typo3conf/ext/tea
+    ├── Resources
+        ├── Private
+        |   ├── Language
+        |   ├── Layouts
+        |   ├── Partials
+        |   └── Templates
+        └── Public
+            ├── CSS
+            ├── Icons
+            ├── Images
+            └── JavaScript
+
+
+:file:`Resources/Private`
+    All ressource files that do not have to be loaded directly by a browser
+    **should** go in this directory. This includes Fluid templating files
```

Comment:
```suggestion
    **must** go in this directory. This includes Fluid templating files.
```

Reply:
Is it? Some people put for example  scss files in the public folder even though they don't have to be accesssible by the browser? Main rule is: When we use **must** we describe what happens otherwise

Reply:
With typo3/cms-composer-installers v4 everything in Resources/Public is accessible by browser. As it is also with v3 as it is not restricted by .htaccess. I would say so, yes. If you want to be more polite say "strongly recommended". I want to emphasize that only accessible artifacts should be placed in Public.
Remark: I also heard from people who placed PHP scripts into Public :-(.

Reply:
There are also installation that have an Assets folder and then some Building processes that put all publicly available files into public. that is also legit...